upb/json/parser.rl

```diff
@@ -882,15 +904,23 @@ static bool end_bool(upb_json_parser *p, bool val) {
 
 static bool start_stringval(upb_json_parser *p) {
   if (is_top_level(p)) {
-    if (!is_string_wrapper_object(p)) {
+    if (is_string_wrapper_object(p)) {
+      start_wrapper_object(p);
+    } else if (is_timestamp_object(p)) {
+      start_object(p);
```

Comment:
Can we have a separate start_timestamp() function that handles all of these timestamp cases? If timestamp is a separate machine it shouldn't call start_stringval() should it?

Reply:
start_stringval prepare the parser for parsing string data. We did the same for string wrapper. I think it makes sense to me because all string, string wrapper and timestamp have string json value. Doing this way also makes the ragel machine easier, otherwise we have to check stuff like whether it's top level message, whether it's timestamp in the ragel machine (basically copy the logic in start_stringval to ragel machine). Do you want me to change this?